services/download/src/main/java/app/coronawarn/server/services/download/FederationBatchProcessor.java

```diff
@@ -165,49 +173,75 @@ private boolean isEfgsEnforceDateBasedDownloadAndNotSeen(String batchTag) {
     LocalDate date = batchInfo.getDate();
     String batchTag = batchInfo.getBatchTag();
-    logger.info("Processing batch for date {} and batchTag {}.", date, batchTag);
+    logger.info("Processing {} batch for date {} and batchTag {}", batchInfo.getSourceSystem(), date, batchTag);
     AtomicReference<Optional<String>> nextBatchTag = new AtomicReference<>(Optional.empty());
     try {
       BatchDownloadResponse response = federationGatewayDownloadService.downloadBatch(batchTag, date);
       AtomicBoolean batchContainsInvalidKeys = new AtomicBoolean(false);
       nextBatchTag.set(response.getNextBatchTag());
       response.getDiagnosisKeyBatch().ifPresentOrElse(batch -> {
-        logger.info("Downloaded {} keys for date {} and batchTag {}.", batch.getKeysCount(), date, batchTag);
+        logger.info("Downloaded {} {} keys for date {} and batchTag {}", batchInfo.getSourceSystem(),
+            batch.getKeysCount(), date, batchTag);
+        Map<String, Integer> countedKeysByOriginCountry = batch
+            .getKeysList().stream().collect(Collectors.groupingBy(
+                app.coronawarn.server.common.protocols.external.exposurenotification.DiagnosisKey::getOrigin))
+            .entrySet().stream()
+            .collect(Collectors.toMap(Entry::getKey,
+                e -> e.getValue().size()));
+        if (config.getSourceSystem() == FederationBatchSourceSystem.EFGS) {
+          countedKeysByOriginCountry.forEach((key, value) -> logger.info("Downloaded {} {} keys with origin country {}",
+              batchInfo.getSourceSystem(), key, value));
+        }
+        if (isChgs()) {
+          countedKeysByOriginCountry.entrySet().stream().filter(k -> !CH.equalsIgnoreCase(k.getKey()))
+              .forEach(k -> logger
+                  .warn("There are keys {} with origin country {} which is different to CH and therefore they will be "
+                      + "dropped", k.getValue(), k.getKey()));
+        }
+
         if (config.isBatchAuditEnabled()) {
           federationGatewayDownloadService.auditBatch(batchTag, date);
         }
         List<DiagnosisKey> validDiagnosisKeys = extractValidDiagnosisKeysFromBatch(batch);
         int numOfInvalidKeys = batch.getKeysCount() - validDiagnosisKeys.size();
         if (numOfInvalidKeys > 0) {
           batchContainsInvalidKeys.set(true);
-          logger.info("{} keys failed validation and were skipped.", numOfInvalidKeys);
+          logger.info("{} {} keys failed validation and were skipped", batchInfo.getSourceSystem(), numOfInvalidKeys);
         }
         int insertedKeys = diagnosisKeyService.saveDiagnosisKeys(validDiagnosisKeys);
-        logger.info("Successfully inserted {} keys for date {} and batchTag {}.", insertedKeys, date, batchTag);
-      }, () -> logger.info("Batch for date {} and batchTag {} did not contain any keys.", date, batchTag));
+        logger.info("Successfully inserted {} {} keys for date {} and batchTag {}", batchInfo.getSourceSystem(),
```

Comment:
Is this log complete ? seems it's expecting 4 arguments but it gets 3.

Reply:
uh, oh... good catch @ioangut - can you please prepare a PR for this on main...